core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -304,4 +322,34 @@ synchronized boolean updateIndexMetaData(IndexMetaData indexMetaData) {
     List<Consumer<Settings>> getUpdateListeners() { // for testing
         return updateListeners;
     }
+
+    /**
+     * Returns the translog durability for this index.
+     */
+    public Translog.Durability getTranslogDurability() {
+        return durability;
+    }
+
+    private Translog.Durability getFromSettings(Settings settings, Translog.Durability defaultValue) {
+        final String value = settings.get(INDEX_TRANSLOG_DURABILITY, defaultValue.name());
+        try {
+            return Translog.Durability.valueOf(value.toUpperCase(Locale.ROOT));
+        } catch (IllegalArgumentException ex) {
+            logger.warn("Can't apply {} illegal value: {} using {} instead, use one of: {}", INDEX_TRANSLOG_DURABILITY, value, defaultValue, Arrays.toString(Translog.Durability.values()));
```

Comment:
oh nevermind I thought it was a parsing method, but this is the simple Enum.valueOf